### PR TITLE
Create video anonymizer

### PR DESCRIPTION
**Pull request by petrabarus**
- [x] Detect subimage
- [x] Censor subimage
- [ ] Read mp4 and censore sub image frame by frame
- [ ] Write acceptance tests

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -77,22 +77,7 @@ task mavenCapsule(type: MavenCapsule){
     capsuleManifest {
         minJavaVersion = '1.8.0'
         dependencyConfiguration configurations.compile
-
-        //noinspection GroovyAssignabilityCheck
-        platform('windows'){
```
Comment (julianghionoiu) [inline build.gradle]: Needed for the video library

File: build.gradle [hunk 2]
```diff
@@ -108,87 +93,11 @@ jacocoTestReport {
...
-    archives sourceJar
-    archives javadocJar
-}
-
-uploadArchives {
-    repositories.mavenDeployer {
-        name = 'bintray-julianghionoiu-dev-screen-anonymise'
-        repository(id: 'bintray-julianghionoiu-dev-screen-anonymise',
-                url: 'https://api.bintray.com/maven/julianghionoiu/maven/dev-screen-anonymise') {
-            authentication(userName: readSystem("BINTRAY_USER"), password: readSystem("BINTRAY_KEY"))
```
Comment (julianghionoiu) [inline build.gradle]: Needed for deployments

**Review by julianghionoiu — COMMENTED** [review]
I think removing files was not a good idea. Most of them will be very useful for video processing.
I think we should just add the ImageMasker and the test on top of master and remove the files once we have the end-to-end acceptance test working.

We will have three acceptance tests:
- One manual acceptance test that anonymises the video from resources
- One test where the input is a stream of barcodes (see existing tests) and obfuscates the barcodes for some chosen numbers
- One performance test

**Comment by julianghionoiu** [comment]
I am ok with the current approach, it is going in the right direction. (FFMpegGrabber)
I think the next step is to write the Automated Acceptance test that describes this functionality and merge it into master.

The automated acceptance test should:
   - read a video made with known barcodes, at least two identical barcodes per frame (see for barcode generation `acceptance.CanRecordVideoTest#can_record_video_at_specified_frame_rate_and_speed` and `tdl.record.image.input.InputFromStreamOfBarcodes#InputFromStreamOfBarcodes`)
   - two barcode subimages
   - run the processing
   - check that the produced video does not contain the subimage barcodes (check `acceptance.CanRecordVideoTest#can_record_video_at_specified_frame_rate_and_speed`)

**Comment by julianghionoiu** [comment]
One other thing, the pixelised images do not look very nice, we should use a heavy Blur instead

**Comment by julianghionoiu** [comment]
After we merge this PR, we should focus extensively on the performance. I will have to process at least 30 hours of videos so each individual step matters.

The key metric is the processing overhead. From the tests I have ran using your sample video, just playing the video (no processing) takes 2s, with processing the whole video takes 8s, that means `400%` overhead.

First we need a performance test. In the same test run, do this
   - compute baseline, run video without any processing
   - compute baselinePlusProcessing, run video with subimages processing
   - compute overhead as percentage: baselinePlusProcessing-baseline/baseline
   - add an assertion on this number, the lower, the better

After reviewing the code, here are a couple of actions we could take to improve performance
 - the matchTemplate returns a Mat with all mins and max, we might be able to read all the maxs and find the location of all matches
 - removeRegionFromImage is not needed if all the matches are gathered in one run (see previous point)
 - do not convert to GREY scale, matchTemplate will do channel compression anyway
 - do not create objects while processing
    - pre-compile blurred images, pre-load subimages into Mats
    - reuse objects (Mat, Frames, Pointer, Buffers), the FFMpeg methods are designed to be used this way
 - no logging during processing, the correct pattern is the one used in `dev-screen-record`, where you only update some stats and another thread does printing
 - Leverage parallelism, the stream that processes subImagePaths can be run in parallel with a reduce operation that combines results.

**Comment by petrabarus** [comment]
Okay, got it. Thanks for the input!

**Comment by petrabarus** [comment]
@julianghionoiu 

> One other thing, the pixelised images do not look very nice, we should use a heavy Blur instead

Okay, I just added blur instead of stamp image.

File: src/test/java/acceptance/CanMaskSubImagesTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                Paths.get(destination),
+                Arrays.asList(new Path[]{subImage1, subImage2})
+        );
+        masker.run();
+
+        List<OutputToBarcodeMatrixReader.TimestampedPayload> tamperedBarcodes
+                = getReadedBarcodeFromVideo(destination).stream()
+                        .filter(CanMaskSubImagesTest::isPayloadOutOfOrder)
+                        .collect(Collectors.toList());
+        assertThat(tamperedBarcodes.size(), is(2));
```
Comment (julianghionoiu) [inline src/test/java/acceptance/CanMaskSubImagesTest.java:67]: Also check the contents of the tampered barcodes, frame number and which on the square should have been masked.

**Review by julianghionoiu — APPROVED** [review]

**Pull request merged**